### PR TITLE
Use ruby2_keywords for delegation

### PR DESCRIPTION
**Pull request by chrisseaton**
### WHY are these changes introduced?

Delegation in Ruby is extremely complicated when you're trying to support a broad range of versions, including future versions.

Using `ruby2_keywords` is the carefully thought-out best-practice way to do this.

* https://eregon.me/blog/2021/02/13/correct-delegation-in-ruby-2-27-3.html
* https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

### WHAT is this pull request doing?

It uses `ruby2_keywords`, via the vendored gem.

There's one very tricky case, which is commented.

### Update checklist

- [ ] ~~I've added a CHANGELOG entry for this PR (if the change is public-facing)~~
- [ ] ~~I've considered possible cross-platform impacts (Mac, Linux, Windows).~~

### CODE REVIEW

File: lib/shopify-cli/method_object.rb [hunk 1]
```diff
@@ -64,10 +64,27 @@ module ClassMethods
...
-            .call(*args, **kwargs.slice(*(kwargs.keys - properties)))
+      ruby2_keywords def call(*args)
+        # This is an extremely complicated case of delegation. The method wants
+        # to delegate arguments, but to have control over which keyword
+        # arguments are delegated. I'm not sure the forward and backward
+        # compatibility of this unusual form of delegation has really been
+        # explored or there's any good way to support it. So I have done
+        # done something hacky here and I'm looking at the last argument and
+        # modifying the package of arguments to be delegated in-place.
+        if args.last.is_a?(Hash)
```
Comment (eregon) [inline lib/shopify-cli/method_object.rb:75]: A way to make sure it's a keyword Hash and not a positional Hash as last argument is `args.last.is_a?(Hash) && Hash.ruby2_keywords_hash?(args.last)`. `Hash.ruby2_keywords_hash?` is also provided by the `ruby2_keywords` gem.

**Review by eregon — COMMENTED** [review]
Looks correct to me.

File: lib/shopify-cli/method_object.rb (hunk 1 above)
Comment (chrisseaton) [reply lib/shopify-cli/method_object.rb:75]: Isn't `Hash.ruby2_keywords_hash? ` always `false`?

Comment (eregon) [reply lib/shopify-cli/method_object.rb:75]: Mmh, but that wouldn't work on Ruby < 2.7, so probably best to keep as-is.

Comment (chrisseaton) [reply lib/shopify-cli/method_object.rb:75]: Context is this gem supports back to 2.5, as it's a tool that may be used by a broad-range of end-users.

**Review by macournoyer — APPROVED** [review]
LGTM.

I see @t6d you created the `MethodObject` class. You want to review this part of the change?

**Review by pepicrft — APPROVED** [review]

**Pull request merged**